src/cascadia/TerminalConnection/CTerminalHandoff.cpp

```diff
@@ -0,0 +1,85 @@
+#include "pch.h"
+
+#include "CTerminalHandoff.h"
+
+// The callback function when a connection is received
+static NewHandoff _pfnHandoff = nullptr;
+// The registration ID of the class object for deregistration later
+static DWORD g_cTerminalHandoffRegistration = 0;
+
+// Routine Description:
+// - Starts listening for TerminalHandoff requests by registering
+//   our class and interface with COM.
+// Arguments:
+// - pfnHandoff - Function to callback when a handoff is received
+// Return Value:
+// - S_OK, E_NOT_VALID_STATE (start called when already started) or relevant COM registration error.
+HRESULT CTerminalHandoff::s_StartListening(NewHandoff pfnHandoff)
+{
+    RETURN_HR_IF(E_NOT_VALID_STATE, _pfnHandoff != nullptr);
+
+    // We could probably hold this in a static...
+    auto classFactory = Make<SimpleClassFactory<CTerminalHandoff>>();

```

Comment:
You didn't put a TODO here. So just wanted to highlight it for you haha

Reply:
thanks, I dropped the comment. I don't want to get too creative here and am just going to leave it be for now.